tests/testthat/test-impact.R

```diff
@@ -123,4 +123,141 @@ test_that("test if vimpact functions are working as expected for central estimat
+  ## years range
+  impact <- impact_by_year_of_vaccination_country_perspective(
+    raw_impact, fvps, "campaign", 2005:2030)
+  expect_equal(impact, data_frame(
+    country = c("ETH", "PAK"),
+    burden_outcome = rep("deaths", 2),
+    value = c(345, 456),
+    fvps = c(23, 98),
+    impact_ratio = c(15.0, 4.6)
+  ), tolerance = 1.e-1)
+
+  expect_error(
+    impact_by_year_of_vaccination_country_perspective(raw_impact, fvps,
+                                                      "routine", 1980:1990),
+    "No FVP data for this range of vaccination years")
+
+  raw_impact$age <- rep(10, 10)
+  expect_error(
+    impact_by_year_of_vaccination_country_perspective(raw_impact, fvps,
+                                                      "routine", 2000:2005),
+    "No impact data for this range of birth cohort and vaccination years")
+
+  expect_error(
+    impact_by_year_of_vaccination_country_perspective(raw_impact, fvps, "test",
+                                                      2000:2030),
+    'Activity type must be "routine" or "campaign" got "test".'
+  )
+})
+
+test_that("impact calculation by year of vaccination cohort perspective", {
+  raw_impact <- data_frame(
+    country = c(rep("ETH", 5), rep("PAK", 5)),
+    year = rep(2001:2005, 2),
+    age = rep(0, 10),
+    time = rep(2001:2005, 2),
+    value = c(234, 456, 345, 234, 345, 934, 567, 876, 675, 456),
+    burden_outcome = rep("deaths", 10)
+  )
+
+  fvps <- data_frame(
+    vaccine = rep("HepB", 15),
+    activity_type = c(rep("routine", 10), rep("campaign", 5)),
+    country = c(rep("ETH", 5), rep("PAK", 10)),
+    year = rep(2001:2005, 3),
+    age = rep(0, 15),
+    fvps = c(34, 54, 34, 54, 23, 65, 78, 98, 78, 98, 43, 45, 65, 45, 65),
+    disease = rep("HepB", 15)
+  )
+
+  impact <- impact_by_year_of_vaccination_cohort_perspective(raw_impact, fvps,
+                                                             2000:2030)
+  expect_equal(colnames(impact),
+              c("country", "birth_cohort", "burden_outcome", "value", "fvps",
+                "impact_ratio"))
+  ## One row for each country, burden_outcome, birth_cohort combination
+  expect_equal(nrow(impact), 10)
+  expect_equal(unique(impact$country), c("ETH", "PAK"))
+  expect_equal(unique(impact$birth_cohort), 2001:2005)
+  expect_equal(impact$value,
+               c(234, 456, 345, 234, 345, 934, 567, 876, 675, 456))
+  ## This is the unchanged fvps - cohort perspective just
+  expect_equal(impact$fvps, c(34, 54, 34, 54, 23, 108, 123, 163, 123, 163))
+  expect_equal(impact$impact_ratio,
+               c(6.9, 8.4, 10.1, 4.3, 15.0, 8.6, 4.6, 5.4, 5.5, 2.8),
+               tolerance = 1e-1)
+
+  ## Impact calculation filters on impact calculations within range of
+  ## birth cohort
+  raw_impact <- raw_impact[raw_impact$year == 2001, ]
+  impact <- impact_by_year_of_vaccination_cohort_perspective(raw_impact, fvps,
+                                                             2000:2030)
+  expect_equal(impact,
```

Comment:
Similarly this edge case is kind of odd - what do we really want to return here? I'm guessing we're not really interested in those rows where impact ratio is NA